service-front/module/Application/src/Services/OpgApiService.php

```diff
@@ -320,8 +320,19 @@ public function updateIdMethodWithCountry(string $uuid, array $data): array
     {
         $url = sprintf("/cases/%s/update-cp-po-id", $uuid);
 
+        $response = $this->makeApiRequest('/identity/details?uuid=' . $uuid);
+        $methodData = [];
+
+        if (array_key_exists('idMethodIncludingNation', $response)) {
+            $methodData = $response['idMethodIncludingNation'];
+        }
+
+        foreach ($data as $key => $value) {
+            $methodData[$key] = $value;
+        }
```

Comment:
I don't really understanding what's going on here. It looks like it's fetching `idMethodIncludingNation` so that it can then set `idMethodIncludingNation`?

Reply:
idMethodIncludingNation is an array containing two pieces of data, the country and the document type. At this point it will have the country defined but not the document type, this is updating the array and then sending it back to the API

Reply:
So if `idMethodIncludingNation` is set in the form, line 327 is irrelevant because it will then get overwritten on line 331.

Is the point that if `idMethodIncludingNation` is already set, but it _isn't_ set on the form, it would be replaced with blank, so line 327 is making sure it's not overwritten?

Reply:
It's a way of ensuring that if either key is set in the new data but not both, the existing key isn't overwritten with a blank. I know it's a bit inelegant, but it's what worked.